File: GomokuAIJavaProject/AIGomoku/RandomAIGomoku.java
package AIGomoku;
import java.util.Random;

public class RandomAIGomoku extends AIGomoku{

	
	private Random rand;
	
	public RandomAIGomoku(int i) {
		super((char) i);
		rand = new Random();
	}

	@Override
	public boolean playMove(BoardGame board) {		
		int x =  rand.nextInt(0,Board.BOARD_SIZE);
		int y =  rand.nextInt(0,Board.BOARD_SIZE);

		while(!board.placeToken(playerNb, x, y)) {
			x =  rand.nextInt(0,Board.BOARD_SIZE);
			y =  rand.nextInt(0,Board.BOARD_SIZE);
		}
		System.out.println("x is " + x +" y is : " + y);
		return board.isFinished(new Position(y,x), playerNb);
	}
	
	
}

File: CPP/src/RandomAIGomoku.cpp
#include "RandomAIGomoku.hpp"

bool RandomAIGomoku::playMove(BoardGame* board)
{
    std::random_device rd;
    static std::uniform_int_distribution<uint> d(0, 15);
    Position* position = new Position(0,0);

    int x =  d(rd);
    int y =  d(rd);
    position->setX(x);
    position->setY(y);
    while(!board->placeToken(playerNb, position)) {
        x =  d(rd);
        y =  d(rd);
        position->setX(x);
        position->setY(y);
    }
    bool finished = board->isFinished(position,playerNb);
    delete (position);
    return finished;
}